include/drivers/pcie/endpoint/pcie_ep.h

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for the PCIe EP drivers.
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2020 Broadcom
+ *
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PCIE_EP_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PCIE_EP_H_
+
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+#include <stdint.h>
+
+enum pcie_ob_mem_type {
+	PCIE_OB_ANYMEM,
+	PCIE_OB_LOWMEM,		/* Within 32-bit address range */
+	PCIE_OB_HIGHMEM,	/* Above 32-bit address range */
+};
+
+enum pci_ep_irq_type {
+	PCIE_EP_IRQ_LEGACY,
+	PCIE_EP_IRQ_MSI,
+	PCIE_EP_IRQ_MSIX,
+};
+
+enum xfer_direction {
+	HOST_TO_DEVICE,		/* Read from Host */
+	DEVICE_TO_HOST,		/* Write to Host */
+};
+
+struct pcie_ep_driver_api {
+	u32_t (*conf_read)(struct device *dev, unsigned int offset);
```

Comment:
and align your types to zephyr types: unsigned int -> u32_t 

Apply that everywhere

Reply:
ACK, in that case, will change return type to s32_t.

Reply:
The data can potentially span the whole range of 0 - 2^31. It would be better to have `int` as return value only, and have an argument to return the actual value.

Reply:
ACK

Reply:
Done.